### PR TITLE
Flow control switch

### PR DESCRIPTION
**Pull request by lindycoder**
It replaces the SwitchTransactional
Auto wraps all methods with auto-connect or auto-transaction
and Connects AFTER locking

### CODE REVIEW

File: netman/core/objects/flow_control_switch.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    @do_not_wrap_with_flow_control
+    def start_transaction(self):
+        if self.wrapped_switch.in_transaction:
+            return
+
+        self.lock.acquire()
+        try:
+            if not self.wrapped_switch.connected:
+               self.wrapped_switch.connect()
+               self.i_connected = True
```
Comment (JoProvost) [inline netman/core/objects/flow_control_switch.py]: May I propose naming this connection_requested_by_start_transaction ?   (or something smaller that speaks for itself ;) )

**Comment by lindycoder** [comment]
@JoProvost comments has been addressed

**Comment by JoProvost** [comment]
LGTM :+1:

**Comment by emmurd** [comment]
Very interesting! LGTM! :+1:

**Pull request merged**